src/react/components/pages/signIn/signInPage.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react"
+import { ISignIn } from "../../../../models/applicationState";
+import SignInForm from "./signInForm";
+import { Route, Redirect } from "react-router-dom";
+import ApiService from "../../../../services/apiService"
+import IAuthActions, * as authActions from "../../../../redux/actions/authActions";
+import { showError } from "../../../../redux/actions/appErrorActions";
+
+
+export interface ISignInPageProps extends React.Props<SignInPage> {
+    actions: IAuthActions;
+}
+
+export interface ISignInPageState {
+    signin: ISignIn;
+}
+
+export default class SignInPage extends React.Component<ISignInPageProps, ISignInPageState> {
+    constructor(props){
+        super(props);
+        this.state = {
+            signin: null,
+        };
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+
+    }
+
+    private async onFormSubmit(signin: ISignIn) {
+        ApiService.loginWithCredentials(signin).then( token => {
+                this.props.actions.signIn(token.data)
```

Comment:
So this action is going to have also the name of the user. Im doing this task right now, and there will be a object:
```
{
    email: string,
   authToken: string,
}
```
So you could implement it on this stage I think :). 

I'm still not sure about if we should have `email` or `username` as a name. Both looks fine for me.